src/Illuminate/Foundation/Auth/ResetsPasswords.php

```diff
@@ -24,8 +25,16 @@
      */
     public function showResetForm(Request $request, $token = null)
     {
+        $email = '';
+        if (is_null($token) == false) {
+            $email = DB::table('password_resets')
+                ->select('email')
+                ->where('token', $token)
+                ->first()->email;
```

Comment:
Looks like a "Trying to get property of non-object" for me on invalid token

Reply:
Oeh, good point! Didn't spot that at first. I think that if I wrap it in a optional (if that helper is available here) that that should cover that use case. Thoughts?